### PR TITLE
Improve publish workflow

### PR DESCRIPTION
**Pull request by miaucl**
Alignment of versioning and prereleases with pypi guidelines: https://pythonpackaging.info/07-Package-Release.html

- add gates for final and prerelease versions
- configure release according to version type
- verify tag version matches with setup and changelog
- add tag to package as `__VERSION__` export

It is only possible to release a final version on the `main` branch. For it to pass the gates, it must have the same version in the `tag`, the `setup.cfg`, the `bring_api/__init__.py` and an entry in the `CHANGELOG.md` file.

To release a prerelease version, no changelog entry is required, **but** it can only happen on a feature branch (**not** main branch). Also, prereleases are marked as such in the github release page.

See https://github.com/miaucl/bring-api/releases/tag/0.8.1a0 for an example of an alpha version testing the new publish workflow.

### CODE REVIEW

File: bring_api/__init__.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Bring API package."""
 
+__VERSION__ = "0.8.1a1"
```
Comment (tr4nt0r) [inline bring_api/__init__.py:3]: i think this should be lowercase

File: setup.cfg [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = bring-api
-version = 0.8.1
+version = 0.8.1a1
```
Comment (tr4nt0r) [inline setup.cfg:3]: This is also possible since setuptools 46.4.0

```suggestion
version = attr: package.__version__

```

**Review by tr4nt0r — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 2 above)
Comment (miaucl) [reply setup.cfg:3]: Then we need to bump `pyproject.toml`, no?
```
[build-system]
requires = [
    "setuptools>=42",
    "wheel"
]
```

File: bring_api/__init__.py (hunk 1 above)
Comment (miaucl) [reply bring_api/__init__.py:3]: ohh, ok. I have it CAPS in all my projects, don't know why exactly. But you are right, can only find lower as well^^
https://stackoverflow.com/questions/458550/standard-way-to-embed-version-into-python-package

File: setup.cfg (hunk 2 above)
Comment (miaucl) [reply setup.cfg]: bumped it to `>=46.4.0`

**Comment by tr4nt0r** [comment]
nice 👍

**Review by tr4nt0r — APPROVED** [review]

**Pull request merged**